test/views/HowGiveInformationViewSpec.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import base.ViewSpecBase
+import fixtures.messages.{HowGiveInformationMessages, WhenReceiveShortageExcessMessages}
+import forms.{HowGiveInformationFormProvider, WhenReceiveShortageExcessFormProvider}
+import models.requests.DataRequest
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.i18n.Messages
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.FakeRequest
+import views.html.{HowGiveInformationView, WhenReceiveShortageExcessView}
+
+import java.time.LocalDate
+
+class HowGiveInformationViewSpec extends ViewSpecBase with ViewBehaviours {
+
+  object Selectors extends BaseSelectors
+
+  "WhenReceiveShortageExcess view" - {
+
+    Seq(HowGiveInformationMessages.English, HowGiveInformationMessages.Welsh).foreach { messagesForLanguage =>
+
+      s"when being rendered in lang code of '${messagesForLanguage.lang.code}'" - {
+
+        implicit val msgs: Messages = messages(app, messagesForLanguage.lang)
+        implicit val request: DataRequest[AnyContentAsEmpty.type] = dataRequest(FakeRequest())
+
+        val form = app.injector.instanceOf[HowGiveInformationFormProvider].apply()
+        val view = app.injector.instanceOf[HowGiveInformationView]
+
+        implicit val doc: Document = Jsoup.parse(view(form, testOnwardRoute).toString())
+
+        behave like pageWithExpectedElementsAndMessages(Seq(
+          Selectors.title -> messagesForLanguage.title,
+          Selectors.h1 -> messagesForLanguage.heading,
+          Selectors.h2(1) -> messagesForLanguage.arcSubheading(testArc),
+          Selectors.radioButton(1) -> messagesForLanguage.whole,
+          Selectors.radioButton(2) -> messagesForLanguage.choose,
+          Selectors.button -> messagesForLanguage.saveAndContinue
```

Comment:
Could add a test for `Selectors.link(1)` which should be the `savePreviousAnswersAndExit` message.